share/ipld/corrupted_data_test.go

```diff
@@ -0,0 +1,41 @@
+package ipld_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/share/availability/full"
+	availability_test "github.com/celestiaorg/celestia-node/share/availability/test"
+	"github.com/celestiaorg/celestia-node/share/service"
+)
+
+// TestNamespaceHasher_CorruptedData is an integration test that verifies that the NamespaceHasher of a recipient of
+// corrupted data will not panic, and will throw away the corrupted data.
+func TestNamespaceHasher_CorruptedData(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	net := availability_test.NewTestDAGNet(ctx, t)
+
+	requestor := full.Node(net)
+	provider, mockBS := availability_test.MockNode(t, net)
+	provider.ShareService = service.NewShareService(provider.BlockService, full.TestAvailability(provider.BlockService))
+	net.ConnectAll()
+
+	// before the provider starts attacking, we should be able to retrieve successfully
+	root := availability_test.RandFillBS(t, 16, provider.BlockService)
+	getCtx, cancelGet := context.WithTimeout(ctx, 2*time.Second)
```

Comment:
Where does the 2s come from? Any timing constants in  testing should have a comment explaining where they came from so that when/if tests become flaky due to timing issues we know which timing constants can be bumped vs debugged.

Reply:
I extracted this into a constant with an explanation that it is arbitrary. But we need to fix this problem node wide, because our tests pass in random time intervals very often.

Reply:
#1273